lib/service/connect.go

```diff
@@ -477,6 +477,12 @@ func (process *TeleportProcess) rotate(conn *Connector, localState auth.StateV2,
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	additionalPrincipals = utils.ReplaceInSlice(
```

Comment:
I'm not sure if `0.0.0.0` is a good thing to be present in principals, is there a way around this?

Reply:
`0.0.0.0` is never added to the list of principals, it's always either removed to replaced. This is similar to how to heartbeat logic works, `0.0.0.0` is used to detect that the client does not know what it's IP is.